### PR TITLE
Structure & traj conversion tools

### PR DESCRIPTION
**Pull request by simonbray**
Built on @tsenapathi's work here: https://github.com/galaxycomputationalchemistry/galaxy-tools-compchem/pull/9

### CODE REVIEW

**Comment by bgruening** [comment]
@simonbray @tsenapathi do you think we should combine both tools?

**Comment by simonbray** [comment]
Either way is possible I guess.

**Comment by simonbray** [comment]
what is your view @bgruening?

**Comment by bgruening** [comment]
My view is as less tools as possible, but the tool should still make sense and should not be overloaded. I can not judge it here but I think here for example it made sense: https://github.com/bgruening/galaxytools/blob/master/chemicaltoolbox/openbabel/ob_convert.xml

**Comment by simonbray** [comment]
I shall combine them :)

File: tools/mdfileconverter/md_converter.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    <description>- interconvert between MD trajectory or structure file formats.</description>
+
+    <requirements>
+        <requirement type="package" version="1.9.1">mdtraj</requirement>
+        <requirement type="package" version="2018.2">gromacs</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        #if $input.input_format == 'gro' or $input.input_format == 'pdb':
```
Comment (bgruening) [inline tools/mdfileconverter/md_converter.xml]: You can use `$input_file.ext` to get the format. With this you could do a single input field without the conditional but wit `format="pdb,trr,xtc"`

File: tools/mdfileconverter/md_converter.xml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            mdconvert ./input.$input.input_format -o ./output.${input.output_format} &&
+            cp ./output.${input.output_format} $output
+        
+        #end if
+
+
+    ]]></command>
+    <inputs>
+        <conditional name="input">
+            <param name="input_format" type="select" label="File format of input">
```
Comment (bgruening) [inline tools/mdfileconverter/md_converter.xml]: If you do not add additional parameters to the single files you could remove the conditional I think, see above.

Comment (bgruening) [reply tools/mdfileconverter/md_converter.xml]: Ah I see, not everything is convertible ....

File: tools/mdfileconverter/md_converter.xml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            
+            <expand macro="traj" format="trr" />
+            <expand macro="traj" format="xtc" />
+            <expand macro="traj" format="dcd" />
+            <expand macro="str" format="pdb" />
+            <expand macro="str" format="gro" />
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="output" auto_format="true" />
```
Comment (bgruening) [inline tools/mdfileconverter/md_converter.xml]: please use here the more explicit form:

https://github.com/galaxyproject/tools-iuc/blob/57493fdc88e6128bed127993464f2a6480e2774d/tools/picrust/macros.xml#L48

File: tools/mdfileconverter/md_converter.xml (hunk 1 above)
Comment (bgruening) [reply tools/mdfileconverter/md_converter.xml]: I see ... leave it like that, sorry for the noise.

**Comment by simonbray** [comment]
@bgruening, I also made some more extensive changes here if you could take a look: https://github.com/simonbray/galaxy-tools-compchem/commit/c3c467fa571bfe8281d7adfd874d6506929b2d26.

Unfortunately the third and fourth tests don't pass, because $input_file.ext returns 'txt' rather than the correct extension (pdb or gro). This isn't a problem for the trr/xtc/dcd extensions. My guess would be something is going wrong with the sniffing, but I don't really know.

**Comment by bgruening** [comment]
@simonbray try adding `ftype="pdb"` to the tests.

**Comment by bgruening** [comment]
Merci!

**Pull request merged**